apis/v1alpha1/clusterprofile_types.go

```diff
@@ -71,16 +71,31 @@ type ClusterProfileStatus struct {
 	// +optional
 	Properties []Property `json:"properties,omitempty"`
 
-	// CredentialProviders is a list credential providers that can provide kubeconfig
-	// credential to connect to the cluster.
+	// CredentialProviders is a list of cluster access providers that can provide access
+	// information for clusters.
+	// Deprecated: Use AccessProviders instead.
```

Comment:
Let's explain what the logic is.
I think the logic could have been simplified (thought it also works this way!).

For ex, we could have ignored CredentialProviders if AccessProviders was specified.

```suggestion
	// Deprecated: Use AccessProviders instead. If both AccessProviders and CredentialProviders are provided, both are used. In case they specify a provider with the same name, the one in AccessProviders is preferred.
```